api/helpers/gamestate/publish-game-state.js

```diff
@@ -0,0 +1,132 @@
+const MoveType = require('../../../utils/MoveType.json');
+
+module.exports = {
+  friendlyName: 'Publish Game State',
+
+  description: 'Determines if the game has ended and sends a socket event based on the game status.',
+
+  inputs: {
+    game: {
+      type: 'ref',
+      description: 'Game overview object',
+      required: true,
+    },
+    gameState: {
+      type: 'ref',
+      description: 'Game state object',
+      required: true,
+    },
+  },
+
+  fn: async function ({ game, gameState }, exits) {
+    try {
+      //Combine game and gamestate users and delete passwords
+      const p0 = { ...game.p0, ...gameState.p0 };
+      delete p0.encryptedPassword;
+      const p1 = { ...game.p1, ...gameState.p1 };
+      delete p1.encryptedPassword;
+      const players = [p0, p1];
+
+      const lastEventTargetType = () => {
+        if (!['targetedOneOff', 'sevenTargetedOneOff'].includes(gameState.moveType)) {
+          return '';
+        }
+        if (gameState.targetCard?.rank === 11) {
+          return 'jack';
+        }
+        if (gameState.targetCard?.rank > 11) {
+          return 'faceCard';
+        }
+        return 'point';
+      };
+
+      const victory = await sails.helpers.gamestate.checkGameStateForWin(game, gameState);
+
+      const countPasses = () => {
+        let numPasses = 0;
+        for (const gameState of game.gameStates.slice(-3)) {
+          if (gameState.moveType !== MoveType.PASS) {
+            return numPasses;
+          }
+          numPasses++;
+        }
+        return numPasses;
+      };
+
+      const fullLog = sails.helpers.gamestate.getLog(game);
+
+      //Last Event, and Extra Socket Variables
+      const happened = gameState.moveType === MoveType.RESOLVE;
+      const { playedBy } = gameState;
+      const discardedCards = gameState.discardedCards.length ? gameState.discardedCards : null;
+      const chosenCard = gameState.moveType === MoveType.RESOLVE_THREE ? gameState.targetCard : null;
+      const pNum = playedBy;
```

Comment:
I'm fine with either way, but if we do go with an object representation, I don't think I like `lastEvent` considering it will also be used for the current move, maybe we could just use `event` If we want to change it